objects: [ box1, box2, box3, box4]
mesh_dir: /home/kinect/JackHadfield/catkin_ws/src/object_assembly_ros/resource/cube30mm24vert.obj

# type of symmetry that characterizes each object. Used to check multiple
# possible configurations. Can be "none", "cubic", "cylindrical" (TODO), spherical (TODO).
# Note that the meshes used to estimate the object poses must have their
# origin set correctly for this to work. Parts with cubic or spherical symmetry
# must have their origin at the center of the mesh. Parts with cylindrical symmetry
# must have their origin on the central axis and the central axis must align with the z axis of it's coordinate frame00.
object_symmetries: [ cubic, cubic, cubic, cubic ] 

use_gui: false

number_of_checks: 1

margin: [0.005, 0.005, 0.005, 1.1, 0.1, 0.1, 0.1] #0.01->slightly over 10 degree difference

number_of_tasks: 3

task1:
  name: castle
  number_of_connections: 3
  connection1:
    parts: [1, 2] #1 based
    relative_pose: [0.03, 0, 0, 0, 0, 0, 1] #part2.x = part1.x + 0.03
  connection2:
    parts: [2, 3]
    relative_pose: [0.03, 0, 0, 0, 0, 0, 1]
  connection3:
    parts: [3, 4]
    relative_pose: [0, 0, -0.03, 0, 0, 0, 1]
task2:  
  name: rail
  number_of_connections: 3
  connection1:
    parts: [1, 2]
    relative_pose: [0.03, 0, 0, 0, 0, 0, 1]
  connection2:
    parts: [2, 3]
    relative_pose: [0.03, 0, 0, 0, 0, 0, 1]
  connection3:
    parts: [4, 3]
    relative_pose: [0.03, 0, 0, 0, 0, 0, 1]
task3:
  name: tower
  number_of_connections: 4
  connection1:
    parts: [1, 2]
    relative_pose: [-0.015, 0.03, 0, 0, 0, 0, 1]
  connection2:
    parts: [1, 3]
    relative_pose: [0.015, 0.03, 0, 0, 0, 0, 1]
  connection3:
    parts: [2, 3]
    relative_pose: [0.03, 0, 0, 0, 0, 0, 1]
  connection4:
    parts: [2, 4]
    relative_pose: [-0.015, 0.03, 0, 0, 0, 0, 1]

